File: app/src/main/java/de/bitb/astroskop/model/converter/ZodiacConverter.java
package de.bitb.astroskop.model.converter;

import de.bitb.astroskop.model.enums.Zodiac;
import io.objectbox.converter.PropertyConverter;

public class ZodiacConverter implements PropertyConverter<Zodiac, Integer> {

    @Override
    public Zodiac convertToEntityProperty(Integer databaseValue) {
        for (Zodiac zodiac : Zodiac.values()) {
            if (zodiac.getId().equals(databaseValue)) {
                return zodiac;
            }
        }
        return null;
    }

    @Override
    public Integer convertToDatabaseValue(Zodiac zodiac) {
        return zodiac.getId();
    }

}

File: app/src/main/java/de/bitb/astroskop/model/converter/ZodiacConverter.kt
package de.bitb.astroskop.model.converter

import de.bitb.astroskop.model.enums.Zodiac
import io.objectbox.converter.PropertyConverter

class ZodiacConverter : PropertyConverter<Zodiac, Int> {

    override fun convertToEntityProperty(databaseValue: Int?): Zodiac? {
        for (zodiac in Zodiac.values()) {
            if (zodiac.id == databaseValue) {
                return zodiac
            }
        }
        return null
    }

    override fun convertToDatabaseValue(zodiac: Zodiac): Int? {
        return zodiac.id
    }

}